src/App/pages/ProfilePage/ProfilePage.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { useEffect, useState } from 'react';
+import { Text, Button, Layout } from 'components';
+import rootStore from 'stores/instance';
+import { observer } from 'mobx-react-lite';
+import { useNavigate } from 'react-router-dom';
+import { Loading } from 'components';
+import RoutesConfig from 'routes';
+import useIsMobile from 'hooks/useIsMobile';
+
+import styles from './ProfilePage.module.scss';
+
+const ProfilePage: React.FC = () => {
+  const { authStore } = rootStore;
+  const navigate = useNavigate();
+
+  const [loading, setLoading] = useState(true);
+  const [imagePreview, setImagePreview] = useState<string | null>(null);
+  const [avatarLoading, setAvatarLoading] = useState(false);
+
+  const isMobile = useIsMobile();
+
+  useEffect(() => {
+    const accessToken = localStorage.getItem('accessToken');
+    if (accessToken) {
+      authStore.fetchProfile().finally(() => setLoading(false));
+    } else {
+      setLoading(false);
+      navigate(RoutesConfig.login);
+    }
```

Comment:
в методе стора уже есть проверка на токен, может возвращать какое-нибудь значение из метода и сократить?
```suggestion
    const res = authStore.fetchProfile().finally(() => setLoading(false));
    if (!res) {
      navigate(RoutesConfig.login);
    }
```